phoenixRest/tests/views/user/avatar.py

```diff
@@ -1,48 +1,21 @@
-def upload_avatar_helper(testapp, username, password, path, x, y, w, h, expected_failure=None):
-    token, refresh = testapp.auth_get_tokens(username, password)
+from test_app import TestApp
 
-    # Get some info about the current user
-    currentUser = testapp.get('/user/current', headers=dict({
-        "Authorization": "Bearer " + token
-        }), status=200).json_body
+NORMAL_AVATAR = "phoenixRest/tests/assets/avatar_test.png"
+TRANSPARENT_AVATAR = "phoenixRest/tests/assets/transparent.png"
 
-    avatar_uuid = currentUser['avatar_uuid']
-    user_uuid = currentUser['uuid']
-    if avatar_uuid is not None:
-        # Delete the avatar so we can test with a new one
-        # This is done so the tests can run locally
-        testapp.delete('/avatar/%s' % avatar_uuid, headers=dict({
-            "Authorization": "Bearer " + token
-            }), status=200)
+def test_upload_avatar(testapp:TestApp):
+    testapp.upload_avatar("test@example.com", "sixcharacters", NORMAL_AVATAR, 10, 10, 600, 450)
 
-    upload_res = testapp.post('/user/%s/avatar' % user_uuid, params="x=%d&y=%d&w=%d&h=%d"% (x, y, w, h), upload_files=[('file', path)], headers=dict({
-        "Authorization": "Bearer " + token
-    }), status = (expected_failure if expected_failure is not None else 200))
+def test_upload_transparent_avatar(testapp:TestApp):
+    testapp.upload_avatar("test@example.com", "sixcharacters", TRANSPARENT_AVATAR, 10, 10, 600, 450)
 
-    if expected_failure is None:
-        upload_res = upload_res.json_body
-        # Try to delete it
-        testapp.delete('/avatar/%s' % upload_res['uuid'], headers=dict({
-            "Authorization": "Bearer " + token
-        }), status=200)
-
-def test_upload_avatar(testapp):
-    # Log in as the test user
-    upload_avatar_helper(testapp, 'test@example.com', 'sixcharacters', 'phoenixRest/tests/assets/avatar_test.png', 10, 10, 600, 450)
-
-def test_upload_transparent_avatar(testapp):
-    # Log in as the test user
-    upload_avatar_helper(testapp, 'test@example.com', 'sixcharacters', 'phoenixRest/tests/assets/transparent.png', 10, 10, 600, 450)
-    
-
-def test_upload_avatar_bad_bounds(testapp):
+def test_upload_avatar_bad_bounds(testapp:TestApp):
     # Ensure the minimum requirements are held
-    upload_avatar_helper(testapp, 'test@example.com', 'sixcharacters', 'phoenixRest/tests/assets/transparent.png', 10, 10, 10, 450, expected_failure=400)
-    upload_avatar_helper(testapp, 'test@example.com', 'sixcharacters', 'phoenixRest/tests/assets/transparent.png', 10, 10, 600, 10, expected_failure=400)
+    testapp.upload_avatar("test@example.com", "sixcharacters", TRANSPARENT_AVATAR, 10, 10, 10, 450, expected_status=400)
```

Comment:
You still need the helper(you refactored too much functionality into testapp)

Reply:
yes, so basically same as before this PR but with the upload part refactored out